File: scripts/gen-favicons.js
// Generate PNG favicons at multiple sizes from the SVG monogram using sharp.
const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

const SVG = '/home/z/my-project/public/brand/monogram.svg';
const OUT_DIR = '/home/z/my-project/public/brand';

const sizes = [16, 32, 48, 64, 96, 128, 180, 192, 256, 512, 1024];
const svgBuffer = fs.readFileSync(SVG);

(async () => {
  for (const s of sizes) {
    const out = path.join(OUT_DIR, `favicon-${s}.png`);
    await sharp(svgBuffer, { density: 384 })
      .resize(s, s, { fit: 'contain', background: { r: 0, g: 0, b: 0, alpha: 0 } })
      .png()
      .toFile(out);
    console.log(`OK  ${s}x${s}  -> ${path.basename(out)}`);
  }

  // favicon.ico (multi-size) — sharp can write an ICO via ico plugin, but we'll just save a 32x32 PNG named favicon.ico
  // Actually let's just save a single 32x32 png and rename to .ico — modern browsers accept PNG-format .ico
  const icoOut = path.join(OUT_DIR, 'favicon.ico');
  await sharp(svgBuffer, { density: 384 })
    .resize(32, 32, { fit: 'contain', background: { r: 0, g: 0, b: 0, alpha: 0 } })
    .png()
    .toFile(icoOut);
  console.log(`OK  favicon.ico (32x32 PNG)`);

  // Apple touch icon (180x180 with white background)
  const appleOut = path.join(OUT_DIR, 'apple-touch-icon.png');
  await sharp(svgBuffer, { density: 384 })
    .resize(180, 180, { fit: 'contain', background: { r: 255, g: 255, b: 255, alpha: 1 } })
    .flatten({ background: '#FFFFFF' })
    .png()
    .toFile(appleOut);
  console.log(`OK  apple-touch-icon.png (180x180)`);

  console.log('\nDone.');
})().catch(e => { console.error('ERR', e); process.exit(1); });

File: scripts/gen-favicons.py
#!/usr/bin/env python3
"""Generate PNG favicons at multiple sizes from the SVG monogram."""
import subprocess
import os
from pathlib import Path

SVG = "/home/z/my-project/public/brand/monogram.svg"
OUT_DIR = Path("/home/z/my-project/public/brand")
OUT_DIR.mkdir(parents=True, exist_ok=True)

sizes = [16, 32, 48, 64, 96, 128, 180, 192, 256, 512, 1024]

for s in sizes:
    out = OUT_DIR / f"favicon-{s}.png"
    try:
        subprocess.run(
            ["rsvg-convert", "-w", str(s), "-h", str(s), "-o", str(out), SVG],
            check=True, capture_output=True
        )
        print(f"OK  {s}x{s}  -> {out.name}")
    except FileNotFoundError:
        try:
            subprocess.run(
                ["convert", "-background", "none", "-density", "384",
                 SVG, "-resize", f"{s}x{s}", str(out)],
                check=True, capture_output=True
            )
            print(f"OK  {s}x{s}  -> {out.name} (ImageMagick)")
        except FileNotFoundError:
            print(f"SKIP {s}x{s}  - no rsvg-convert or convert found")

ico_out = OUT_DIR / "favicon.ico"
try:
    subprocess.run(
        ["convert", "-background", "none",
         str(OUT_DIR / "favicon-16.png"),
         str(OUT_DIR / "favicon-32.png"),
         str(OUT_DIR / "favicon-48.png"),
         str(OUT_DIR / "favicon-64.png"),
         str(ico_out)],
        check=True, capture_output=True
    )
    print(f"OK  favicon.ico")
except Exception as e:
    print(f"WARN ico conversion failed: {e}")

apple = OUT_DIR / "apple-touch-icon.png"
try:
    subprocess.run(
        ["convert", "-background", "#FFFFFF", "-gravity", "center",
         "-extent", "180x180", str(OUT_DIR / "favicon-180.png"), str(apple)],
        check=True, capture_output=True
    )
    print(f"OK  apple-touch-icon.png")
except Exception as e:
    print(f"WARN apple icon failed: {e}")

print("\nFiles in:", OUT_DIR)
for f in sorted(OUT_DIR.iterdir()):
    print(f"  {f.name}  ({f.stat().st_size} bytes)")
